### PR TITLE
Remove Duplicate in a Linked List Without using  Extra Space

### PR DESCRIPTION
**Pull request by sumitgsh**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
Yes
### Description
Removes Duplicates From a linked list without using extra storage
(Write your answer here.)

### Checklist

- [X] I've read the contribution guidelines.
- [X] I've checked the issue list before deciding what to submit.
- [X] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
#2357

### CODE REVIEW

**Comment by sumitgsh** [comment]
@HarshCasper @siddharth25pandey @iamrajiv please review my PR

**Comment by sumitgsh** [comment]
@HarshCasper I was having some conflict with my previous branch #3285 while I was trying to update changes that ur requested.
So I closed the above branch and created a new branch #3357 with the changes that u requested. Sorry for the inconvenience.

File: Java/README.md [hunk 1]
```diff
@@ -120,6 +120,7 @@ _add list here_
 - [Priority Queue](ds/PriorityQueueEg.java)
 - [AVL Tree](ds/AVL_Tree.java)
 - [Balanced Brackets](ds/balancedBrackets.java)
+- [Remove Duplicate In a Unsorted Linked list(No Extra Space Allowed)](ds/RemDupInLLWithoutExtraSpace.java)
```
Comment (ankitaggarwal23) [inline Java/README.md]: Make your title short, concise and readable.

File: Java/ds/RemDupInLLWithoutExtraSpace.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
+/* Java Program to Remove Duplicates from a Unsorted Linked List 
+Without using any Extra Space i.e while finding Duplicates we can't
+use any Data Structure to store values.
+Then we can use two pointers approach :current and runner ,current iterates through the linked list
+and runner check all subsequent nodes for every current and try to find duplicates in the list.
+*/
```
Comment (ankitaggarwal23) [inline Java/ds/RemDupInLLWithoutExtraSpace.java]: Use multiline comments `//` for description

File: Java/ds/RemDupInLLWithoutExtraSpace.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+import java.util.*;
+
+public class RemDupInLLWithoutExtraSpace {
```
Comment (ankitaggarwal23) [inline Java/ds/RemDupInLLWithoutExtraSpace.java:9]: Add a one line description of what this functions is.

File: Java/ds/RemDupInLLWithoutExtraSpace.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  static Node head;
+  class Node {
+    public int data;
+    public Node next;
+    Node(int data) {
+      this.data = data;
+    }
+  }
+
+  public void insert_At_beginning(int data) {
```
Comment (ankitaggarwal23) [inline Java/ds/RemDupInLLWithoutExtraSpace.java:20]: Add a one line description of what this functions is.

File: Java/ds/RemDupInLLWithoutExtraSpace.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+      head = ptr;
+    }
+  }
+  //check whether a Linked List is empty or not
+  public boolean isEmpty() {
+    if (head == null)
+      return true;
+    return false;
+  }
+  public void display() {
```
Comment (ankitaggarwal23) [inline Java/ds/RemDupInLLWithoutExtraSpace.java:38]: -Add a new line above display function

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by sumitgsh** [comment]
@ankitaggarwal23 can u please review my changes

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**